### ISSUE TITLE
Ndots = 1

### ISSUE THREAD

**Issue by eBeyond** [miekg/dns#586]
Hi within my docker swarm cluster the resolv.conf is configured using options ndots0. It's required for service discovery. Within your lib you set ndots always to 1. As such I can't use the library for name resolution, it's also used by prometheus and as such it's also not usable in the cluster. :(
Images in docker swarm mode needs this setting to discover the services. When set to > 0 then the resolution didn't work.

**Comment by miekg** [comment]
clientconfig.go indeed has:
~~~
					if n < 1 {
						n = 1
					}
~~~
I don't understand the problem...

> "The default for n is 1, meaning that if there are any dots in a name, the name will be tried first as an > absolute name before any search list elements are appended to it."

But it seems ndots:0 is an allowed setting

**Comment by miekg** [comment]
Go std library also allows it (now). Code was copied from there in the go 0.6 era. Care to send a PR ?

**Comment by miekg** [comment]
~~~
					n, _, _ := dtoi(s[6:])
					if n < 0 {
						n = 0
					} else if n > 15 {
						n = 15
					}
					conf.ndots = n
~~~
is current std lib code.

**Comment by eBeyond** [comment]
I can send you an PR on monday. ;)

**Comment by eBeyond** [comment]
I've filed a PR for that. Thanks

**Comment by eBeyond** [comment]
I don't think that the commit fixes this error.

**Comment by miekg** [comment]
? Which commit and what error?

**Comment by eBeyond** [comment]
There is a notice above my comment about a commit to kokizzu/dns. Within the commitmessage there is a "Fixes 586" but it doesn't. ;)

**Comment by miekg** [comment]
Ah damn, wrong number used there :/

**Comment by kokizzu** [comment]
my bad, i think my github auto-fetch script pushed this '__')

**Comment by miekg** [comment]
Fixed via #590
